apix-impl/src/main/java/eu/xenit/apix/alfresco/comments/CommentService.java

```diff
@@ -0,0 +1,134 @@
+
+    public CommentService(org.alfresco.repo.forum.CommentService commentService, IContentService contentService,
+            INodeService nodeService, IPermissionService permissionService, ApixToAlfrescoConversion apixConverter) {
+        this.commentService = commentService;
+
+        this.contentService = contentService;
+        this.nodeService = nodeService;
+        this.permissionService = permissionService;
+        this.apixConverter = apixConverter;
+    }
+
+    @Override
+    public NodeRef getTargetDocumentForComment(NodeRef commentNodeRef) {
+        return apixConverter.apix(commentService.getDiscussableAncestor(apixConverter.alfresco(commentNodeRef)));
+    }
+
+    @Override
+    public Conversation getComments(NodeRef targetNode, int skipCount, int pageSize) {
+        org.alfresco.service.cmr.repository.NodeRef alfTargetNode = apixConverter.alfresco(targetNode);
+        PagingResults<org.alfresco.service.cmr.repository.NodeRef> commentAlfNodes =
+                commentService.listComments(alfTargetNode, new PagingRequest(skipCount, pageSize));
+        return new Conversation(
+                commentAlfNodes.getPage().stream().map(alfCommentNode -> toComment(alfTargetNode, alfCommentNode))
+                        .collect(Collectors.toList()),
+                commentAlfNodes.hasMoreItems(),
+                canCreateComment(targetNode));
+    }
+
+    @Override
+    public Comment addNewComment(NodeRef targetNode, String content) {
+        org.alfresco.service.cmr.repository.NodeRef alfTargetNode = apixConverter.alfresco(targetNode);
+        return toComment(alfTargetNode, commentService.createComment(alfTargetNode, "", content, false));
+    }
+
+    @Override
+    public Comment updateComment(NodeRef targetCommentNode, String content) {
+        return updateComment(getTargetDocumentForComment(targetCommentNode), targetCommentNode, content);
+    }
+
+    @Override
+    public Comment updateComment(NodeRef targetDocumentNode, NodeRef targetCommentNode, String content) {
+        org.alfresco.service.cmr.repository.NodeRef alfTargetCommentNode = apixConverter.alfresco(targetCommentNode);
+        commentService.updateComment(alfTargetCommentNode, "", content);
+        org.alfresco.service.cmr.repository.NodeRef alfTargetDocumentNode = apixConverter.alfresco(targetDocumentNode);
+        return toComment(alfTargetDocumentNode, alfTargetCommentNode);
+    }
+
+    @Override
+    public void deleteComment(NodeRef targetCommentNode) {
+        commentService.deleteComment(apixConverter.alfresco(targetCommentNode));
+    }
+
+    @Override
+    public boolean canCreateComment(NodeRef targetNodeRef) {
+        return permissionService.hasPermission(targetNodeRef, IPermissionService.CREATE_CHILDREN);
+    }
+
+    protected Comment toComment(org.alfresco.service.cmr.repository.NodeRef documentNode,
+            org.alfresco.service.cmr.repository.NodeRef commentNodeRef) {
```

Comment:
I would expect most of this method to be part of a constructor for `Comment`. The alf→apix conversion and IOUtils stuff is probably out of scope, but the rest seems like it's behaviour that deals with the state of `Comment`, not the state of `CommentService`, so it should go in `Comment`.

Reply:
im not a fan of putting the disentanglement of the properties from the property map in the pojo class, since the properties are extracted based on QName which is strictly alfresco, while the apix interface should not be dependant on alfresco. The alternative would be a builder, but seeing as I wouldnt want that in the interface either, it would end up in the impl module again.